### PR TITLE
Fix SignTransactionSheet not disabling button on authorization

### PR DESCRIPTION
**Pull request by walmat**
Fixes APP-1969

## What changed (plus any additional context for devs)
Disables sign transaction button once the user prompts for authentication and enables again afterward

## Screen recordings / screenshots
n/a

## What to test
happy paths for message signing, txn signing, etc.

### CODE REVIEW

File: src/components/DappBrowser/BrowserTab.tsx [hunk 1]
```diff
@@ -248,6 +248,7 @@ const FreezableWebViewComponent = ({
         }
         // eslint-disable-next-line no-empty
       } catch (e) {
+        console.log(JSON.stringify(e, null, 2));
```
Comment (walmat) [inline src/components/DappBrowser/BrowserTab.tsx:251]:
```suggestion
```

File: src/hooks/useConfirmTransaction.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ export const useConfirmTransaction = ({
   handleConfirmTransaction,
 }: UseConfirmTransactionProps) => {
   const onConfirm = useCallback(async () => {
+    console.log('onConfirm: ', isMessageRequest, isBalanceEnough, isValidGas);
```
Comment (walmat) [inline src/hooks/useConfirmTransaction.ts:19]:
```suggestion
```

File: src/hooks/useSubmitTransaction.ts [hunk 3]
```diff
@@ -30,19 +33,22 @@ export const useTransactionSubmission = ({
...
     }
-  }, [isAuthorizing, onConfirm]);
+  }, [isAuthorizing, onConfirm, setIsAuthorizing]);
 
   const submitFn = useCallback(
     () =>
       performanceTracking.getState().executeFn({
         fn: async () => {
+          console.log('submitFn: ', isBalanceEnough);
+
```
Comment (walmat) [inline src/hooks/useSubmitTransaction.ts:43]:
```suggestion
```

File: src/hooks/useSubmitTransaction.ts [hunk 4]
```diff
@@ -30,19 +33,22 @@ export const useTransactionSubmission = ({
...
           if (!isBalanceEnough) {
             navigate(Routes.ADD_CASH_SHEET);
             return;
           }
           if (accountInfo.isHardwareWallet) {
             navigate(Routes.HARDWARE_WALLET_TX_NAVIGATOR, { submit: onPressSend });
           } else {
+            console.log('submitFn: sending');
```
Comment (walmat) [inline src/hooks/useSubmitTransaction.ts:51]:
```suggestion
```

File: src/languages/en_US.json [hunk 5]
```diff
@@ -2908,6 +2908,7 @@
         "buttons": {
           "cancel": "Cancel",
           "confirm": "􀎽 Confirm",
+          "confirming": "Confirming",
```
Comment (walmat) [inline src/languages/en_US.json:2911]: Do we want a different copy here?

File: src/screens/SignTransactionSheet.tsx [hunk 6]
```diff
@@ -473,6 +475,7 @@ export const SignTransactionSheet = () => {
   ]);
 
   const handleSignMessage = useCallback(async () => {
+    console.log('handleSignMessage: called');
```
Comment (walmat) [inline src/screens/SignTransactionSheet.tsx:478]:
```suggestion
```

File: src/screens/SignTransactionSheet.tsx [hunk 7]
```diff
@@ -550,18 +550,56 @@ export const SignTransactionSheet = () => {
...
+
+    return colors.alpha(color, canPressConfirm ? 1 : 0.6);
+  }, [colors, simulationResult?.simulationError, simulationResult?.simulationScanResult, canPressConfirm]);
+
   const onPressCancel = useCallback(() => onCancel(), [onCancel]);
 
   const expandedCardBottomInset = EXPANDED_CARD_BOTTOM_INSET + (isMessageRequest ? 0 : GAS_BUTTON_SPACE);
 
-  const canPressConfirm = isMessageRequest || (!!walletBalance?.isLoaded && !!chainId && !!selectedGasFee?.gasFee?.estimatedFee);
+  simulationResult?.simulationError && console.log(JSON.stringify(simulationResult?.simulationError, null, 2));
```
Comment (walmat) [inline src/screens/SignTransactionSheet.tsx:602]:
```suggestion
```

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/APP-1969/5f60ceb8a1e5d7cf223dd989e2f53abc62e17531.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/APP-1969/5f60ceb8a1e5d7cf223dd989e2f53abc62e17531.ipa) for 5f60ceb8a1e5d7cf223dd989e2f53abc62e17531

**Review by derHowie — APPROVED** [review]

**Review by benisgold — APPROVED** [review]

**Pull request merged**